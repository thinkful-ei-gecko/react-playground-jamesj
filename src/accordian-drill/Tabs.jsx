import React from 'react'

export default class Tabs extends React.Component {
  static defaultProps = { tabs: [] }
  render() {
    const buttons = this.props.tabs.map((tab, index) => (
      <button key={index}>{tab.name}</button>
    ))
    const currentTab = this.props.tabs[0]
    return (
      <div>
        {buttons}
          {this.props.tabs.length && (
            <div>
              {currentTab.content}
            </div>
          )}
      </div>
    )
  }
}
