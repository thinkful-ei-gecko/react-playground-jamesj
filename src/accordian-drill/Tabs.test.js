import React from 'react';
import ReactDOM from 'react-dom';
import Tabs from './Tabs';

describe('Tabs component', () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Tabs />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})