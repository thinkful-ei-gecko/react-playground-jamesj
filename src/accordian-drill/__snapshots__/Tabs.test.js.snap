// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders first tab by default 1`] = `
<div>
  <button>
    First tab
  </button>
  <button>
    Second tab
  </button>
  <button>
    Third tab
  </button>
  <div>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.
  </div>
</div>
`;
